### PR TITLE
Unify MIDI note calculation with the audio feature (from #21496)

### PR DESCRIPTION
**Pull request by Nebuleon**
## Description

In #21496, I proposed a change to the note frequency calculation of the audio feature, and in a comment on that pull request, @sigprof pointed out that the MIDI code used a different calculation that was inaudibly imprecise, and distinct from the implementation used in the audio feature.

This pull request:
* unifies the calculation of the note frequency in the MIDI code to the existing code of the audio feature, calculating note pitches based on standard A at exactly 440 Hz;
* changes the calculation so that it's done on `float` rather than `double`, and using `powf` rather than `pow`, like #21496. Like that pull request, this one saves around 4.5 KB of firmware size on ARM with a float-precision FPU (like the STM32F303).

Tested on my Moonlander by playing a MIDI file with the current code and the proposed code, recording that and using a spectrum analyzer, as well as by printing out the frequencies to the console. The difference is there, but is indeed inaudible.

I did not implement this by calling `float compute_freq_for_midi_note(uint8_t)` from the audio feature due to separation of concerns; due to #21496 not having been merged in yet; and because, as noted in #21496, the calculation is different by one octave, and calling `compute_freq_for_midi_note(note + 12)` would cause overflow in argument type conversion for notes 244 to 255 inclusive.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
@Nebuleon was there anything else needed for this PR (since the other two are merged)?

**Comment by Nebuleon** [comment]
The only other thing I can think of is switching out the calculation for a call to `float compute_freq_for_midi_note(uint8_t)` from the audio feature, but I still don't think it's useful to do that despite #21496 being merged because of separation of concerns and overflows.

Now that I think about it, there's also the fact that the MIDI feature does not use the configurable `PITCH_STANDARD_A` but the audio feature does, so changing out the computation would also make MIDI depend on `PITCH_STANDARD_A`, and that would need to be documented.

In summary, I don't think anything else is needed for this one, and it's ready to go. Thanks for checking in!

**Comment by drashna** [comment]
Awesome, thanks!
